sdk/rust/oak/src/grpc/mod.rs

```diff
@@ -155,24 +155,37 @@ impl<T: ServerNode> crate::Node<Invocation> for T {
     /// to be written to.
     ///
     /// [`invoke`]: ServerNode::invoke
-    fn handle_command(
-        &mut self,
-        invocation: Invocation,
-    ) -> std::result::Result<(), crate::OakError> {
+    fn handle_command(&mut self, invocation: Invocation) -> std::result::Result<(), OakError> {
+        let response_writer = ChannelResponseWriter::new(invocation.response_sender);
+
         // Read a single encapsulated request message from the read half.
-        let req: GrpcRequest = invocation.request_receiver.receive()?;
+        let req: GrpcRequest = match invocation.request_receiver.receive() {
+            Ok(req) => req,
+            Err(err) => {
+                // TODO(#1078): If it's clear that the issue is caused by a too restrictive label
+                // (permission denied), then return a more specific error to the caller.
+                warn!(
+                    "could not receive gRPC request from gRPC invocation: {}",
+                    err
+                );
+                // If anything fails while reading the request, then at least try to send a response
+                // back to the caller, rather than just returning from this method with `?`, since
+                // otherwise that would leave the caller hanging forever.
+                return response_writer.close(Result::Err(build_status(
```

Comment:
This seems a bit weird. Even though an error happened, the caller will receive an `Ok` status if the channel closes successfully.

Reply:
You are right, that was not great, I made it so that the result of the closing is just logged, but the original error is returned to the caller. Thanks for the suggestion!